### ISSUE TITLE
 RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by uittorio** [uittorio/ts-auto-mock#85]
When extending an interface that has this in one of his type the compiler report:
 RangeError: Maximum call stack size exceeded

```ts 
interface TestB {
  property: this
}
interface Test extends TestB {}
```

after an investigation I've found out the the "this" reference refer to TestB instead of Test.
That is wrong because in javascript "this" should refer to Test when extending

**Comment by uittorio** [comment]
#83

**Comment by uittorio** [comment]
This issue has been resolved by #87
